bundles/automation/org.eclipse.smarthome.automation.provider.file/src/main/java/org/eclipse/smarthome/automation/internal/provider/file/ModuleTypeFileProvider.java

```diff
@@ -49,8 +60,18 @@ protected String getUID(ModuleType providedObject) {
     public <T extends ModuleType> Collection<T> getModuleTypes(Locale locale) {
         Collection<ModuleType> values = providedObjectsHolder.values();
         if (values.isEmpty()) {
-            return Collections.<T>emptyList();
+            return Collections.<T> emptyList();
         }
         return (Collection<T>) new LinkedList<ModuleType>(values);
     }
+
+    @Override
+    protected boolean checkExistence(String uid) {
+        if (moduleTypeRegistry.get(uid) != null) {
+            logger.error("Module Type with UID \"{}\" already exists! Failed to create a second with the same UID!",
```

Comment:
Having an error logging here is very unexpected. A method `checkExistence` should simply return true or false and not have any side effects.